lib/factory.coffee

```diff
@@ -37,11 +37,15 @@ emit = ($item, item) ->
           <li><a class="menu" href="#" title="#{info.title}">#{info.name}</a></li>
         """
     menu.find('a.menu').click (evt)->
-      $item.removeClass('factory').addClass(item.type=evt.target.text.toLowerCase())
+      type = evt.target.text.toLowerCase()
+      $item.removeClass('factory').addClass(item.type=type)
       $item.unbind()
       evt.preventDefault()
       active.set $item.parents(".page")
-      editor.textEditor $item, item
+      wiki.getPlugin(type, (plugin) ->
```

Comment:
This async operation introduces a visible delay responding to the factory selection. This is an awkward time to be making this decision.

One possible alternative would be to annotate the factories.json info that is loaded at client-start. The server assembles this file from available plugins.

Along this line, the plugin could offer a page of "Managing Slack Histories" along with the usual About Slackmatic Plugin. If this interface were advertised in factory.json in the `pages:[]` field then the hamburger menu would offer this more descriptive option.

You could convey special behavior in selected instances of Slackmatic by just annotating the item json with extra fields configured by text editing pages/manage-slack-histories.json. Here is an example of this in the (Local) Changes plugin. https://github.com/fedwiki/wiki-plugin-changes/blob/master/pages/submit-changes#L35

Reply:
Yes, this PR is focused on the behavior of factory plugin. The Slackmatic plugin is quite unfinished and provided here as an example of a custom editor.

The factory plugin presents a particular challenge. In order to prevent the visible delay, the custom editor must be loaded by the wiki-client before the author chooses that plugin from the factory. The asynchronous wiki.getPlugin() was the first solution that could possibly work.

I like the suggestion to use factories.json to indicate a custom editor.

I was just looking at how the hamburger menu gets populated:
This code in the client:
https://github.com/fedwiki/wiki-client/blob/master/lib/legacy.coffee#L290-L310
...gets data from this endpoint in the server:
https://github.com/fedwiki/wiki-server/blob/master/lib/server.coffee#L329-L347

I think I'll update this PR to pre-load plugins that declare custom editors in factory.json:
https://github.com/fedwiki/wiki-client/blob/master/lib/legacy.coffee#L20-L21

```
wiki.origin.get 'system/factories.json', (error, data) ->
  window.catalog = data
  window.catalog
    .filter((entry) -> entry.customEditor)
    .forEach(entry -> 
      wiki.getPlugin(entry.name.toLowerCase(), plugin ->
        console.log("#{entry.name} Plugin declares a custom editor, so pre-loading the plugin")
      )
    )
```